src/gradualizer_db.erl

```diff
@@ -279,17 +280,29 @@ autoimport(M, #state{opts = #{autoimport := true},
     end.
 
 import_module(Mod, State) ->
-    case State#state.srcmap of
+    case State#state.beammap of
         #{Mod := Filename} ->
             State1 = import_files([Filename], State),
             {ok, State1};
         _ ->
-            not_found
+          case State#state.srcmap of
+              #{Mod := Filename} ->
+                  State1 = import_files([Filename], State),
+                  {ok, State1};
+              _ ->
+                  not_found
+          end
     end.
 
 import_files([File | Files], State) ->
-    EppOpts = [{includes, guess_include_dirs(File)}],
-    {ok, Forms} = epp:parse_file(File, EppOpts),
+    {ok, Forms} =
+        case re:run(File, beam_file_regexp()) of
```

Comment:
I think it is better to make two separate functions for importing from an .erl file and importing a .beam file. It is less logic and the import function doesn't need to use a regex to find out what kind of file it is importing, which is something we already know where the function is called.